lib/jormungandr/exe/cardano-wallet-jormungandr.hs

```diff
@@ -273,7 +276,25 @@ genesisHashOption = optionT $ mempty
     <> help "Blake2b_256 hash of the genesis block, in base 16."
 
 -- | -- [ARGUMENTS...]
-extraArguments :: Parser [Text]
-extraArguments = some $ argument str $ mempty
+extraArguments :: Parser [String]
+extraArguments = many $ argument jmArg $ mempty
     <> metavar "[-- ARGUMENTS...]"
     <> help "Extra arguments to be passed to jormungandr."
+  where
+    jmArg = do
+        arg <- readerAsk
+        case validate arg of
+            Just err -> readerError err
+            Nothing -> pure arg
+    validate arg
+        | "--genesis-block" `isPrefixOf` arg = Just $
+            "The " <> arg <> " option must be placed before the --"
+        | "--rest-listen" `isPrefixOf` arg = Just $
+            suggestion "--rest-listen"
+        | "--storage" `isPrefixOf` arg = Just $
+            suggestion "--storage"
+        | otherwise = Nothing
+    suggestion arg = "The " <> arg <> " argument is used by "
+        <> "`cardano-wallet-jormungandr launch'."
+        <> "\nIf you need this level of flexibility, run `jormungandr' "
+        <> " separately and use `cardano-wallet-jormungandr serve'."
```

Comment:
There are two spaces here (between `'jormungandr'` and `separately`). Also _`jormungandr'_ uses two different quote signs...

Reply:
They are open and close quotes. But I have changed it to symmetrical double quotes and fixed the space.